dev/docker-compose.yml

```diff
@@ -0,0 +1,96 @@
+version: "3"
+
+services:
+  lakevision:
+    image: localhost/lakevision
+    container_name: lakevision
+    build: ../
+    networks:
+      iceberg_net:
+    depends_on:
+      - rest
+      - minio
+    volumes:
+      - ./warehouse:/warehouse
+    environment:
+      - PYICEBERG_CATALOG__DEFAULT__URI=http://rest:8181
+      - PYICEBERG_CATALOG__DEFAULT__S3__ENDPOINT=http://minio:9000
+      - PYICEBERG_CATALOG__DEFAULT__PY_IO_IMPL=pyiceberg.io.fsspec.FsspecFileIO
+      - AWS_ENDPOINT=http://minio:9000
+      - AWS_ACCESS_KEY_ID=admin
+      - AWS_SECRET_ACCESS_KEY=password
+      - AWS_REGION=us-east-1
+    ports:
+      - 8501:8501
+    links:
+      - rest:rest
+      - minio:minio
+  rest:
+    image: tabulario/iceberg-rest
+    container_name: pyiceberg-rest
+    networks:
+      iceberg_net:
+    ports:
+      - 8181:8181
+    environment:
+      - AWS_ACCESS_KEY_ID=admin
+      - AWS_SECRET_ACCESS_KEY=password
+      - AWS_REGION=us-east-1
+      - CATALOG_WAREHOUSE=s3://warehouse/
+      - CATALOG_IO__IMPL=org.apache.iceberg.aws.s3.S3FileIO
+      - CATALOG_S3_ENDPOINT=http://minio:9000
+  minio:
+    image: minio/minio
+    container_name: pyiceberg-minio
+    environment:
+      - MINIO_ROOT_USER=admin
+      - MINIO_ROOT_PASSWORD=password
+      - MINIO_DOMAIN=minio
+    networks:
+      iceberg_net:
+        aliases:
+          - warehouse.minio
+    ports:
+      - 9001:9001
+      - 9000:9000
+    command: ["server", "/data", "--console-address", ":9001"]
+  mc:
+    depends_on:
+      - minio
+    image: minio/mc
+    container_name: pyiceberg-mc
+    networks:
+      iceberg_net:
+    environment:
+      - AWS_ACCESS_KEY_ID=admin
+      - AWS_SECRET_ACCESS_KEY=password
+      - AWS_REGION=us-east-1
+    entrypoint: >
+      /bin/sh -c "
+      until (/usr/bin/mc config host add minio http://minio:9000 admin password) do echo '...waiting...' && sleep 1; done;
+      /usr/bin/mc mb minio/warehouse;
+      /usr/bin/mc policy set public minio/warehouse;
+      tail -f /dev/null
+      "
+  loader:
+    depends_on:
+      - minio
+      - rest
+    image: dataloader
```

Comment:
I had to change the image to `localhost/dataloader`.  Otherwise, you get an error:
```
Getting image source signatures
Copying blob sha256:c1792ed6487b3787130e2ff739c7b090210c326431bd55ece8cd3f932e6e0833
Copying blob sha256:ad3cd4190b2a281e6c57db5b250c359ec7859e0198e930433d18f37c41488b42
Copying blob sha256:a4ed047ad50440da913ed5fd1086a80af17cf896d99a7f6e36d3496508164fe1
Copying blob sha256:0db59e636d467747960ed6ab401c0d7654ee656d56d4170642eb43d6d1f1308b
Copying blob sha256:8e63b938b809f6e08756c0536d0e04a051a31d53a9edbb96446b4844419418f2
Copying blob sha256:d24233d32ae0642206b090d5458e470ed9edc024d3b869aa7169c79241cecf02
Copying config sha256:e4d1058685dfc7b6468550d2bf02b7da34756295ec4fb1bae2209fef36d9b830
Writing manifest to image destination
94dab8795ac9d6c2697d732c12ff18e99a82dfb115fb59149b821f64a09b11e8
634c84fb3349490665c562c4e07bea79079ce78cc9262780879cdafb921a23d7
Resolving "dataloader" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/library/dataloader:latest...
Error: initializing source docker://dataloader:latest: reading manifest latest in docker.io/library/dataloader: requested access to the resource is denied
Error: no container with name or ID "lakevision-dl" found: no such container
```
Same happened to @juancappi